homeassistant/loader.py

```diff
@@ -662,6 +705,8 @@ def __init__(
             self._all_dependencies_resolved = True
             self._all_dependencies = set()
 
+        preload_platforms: list[str] = hass.data[DATA_PRELOAD_PLATFORMS]
+        self._preload_platforms = preload_platforms
```

Comment:
What's the benefit storing this reference on `self` instead of fetching it when needed inside `get_component` ?

Reply:
Don't have to lookup `hass` `data` `DATA_PRELOAD_PLATFORMS` each time and it makes everything a lot more compact since we don't write it out everywhere

Reply:
It's still done as many times right? It's done either in the constructor or inside `get_component`, which we call only once ? 

Ok on it being more compact 👍